### PR TITLE
Config file changes

### LINKED ISSUE #371: DOI Service not logging into Datacite to import data
## 🐛 Describe the bug identified during I&T
Regression tests started failing at build 82. They continued failing with subsequent releases of `doi-service`. Basically, the errors involve login problems to DataCite: 
```
...
Field Map 20220108 Bundle
Traceback (most recent call last):
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/pds_doi_service/core/outputs/web_client.py", line 89, in _submit_content
    response.raise_for_status()
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.test.datacite.org/dois

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/pds_doi_service/core/actions/reserve.py", line 253, in run
    output_doi, o_doi_label = self._web_client.submit_content(
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/pds_doi_service/core/outputs/datacite/datacite_web_client.py", line 81, in submit_content
    response_text = super()._submit_content(
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/pds_doi_service/core/outputs/web_client.py", line 95, in _submit_content
    raise WebRequestException(
pds_doi_service.core.entities.exceptions.WebRequestException: DOI submission request to DataCite service failed, reason: 404 Client Error: Not Found for url: https://api.test.datacite.org/dois
Details: ('{"errors":[{"status":"404","title":"The resource you are looking for '
 'doesn\'t exist."}]}')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/bin/pds-doi-cmd", line 8, in <module>
    sys.exit(main())
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/pds_doi_service/core/cmd/pds_doi_cmd.py", line 42, in main
    output = action.run(**kwargs)
  File "/data/jenkins/workspace/nsole-regression-tests_develop_2/venv/lib/python3.9/site-packages/pds_doi_service/core/actions/reserve.py", line 288, in run
    raise CriticalDOIException(f"{error_message}.  {explanatory_message}")
pds_doi_service.core.entities.exceptions.CriticalDOIException: DOI submission request to DataCite service failed, reason: 404 Client Error: Not Found for url: https://api.test.datacite.org/dois
Details: ('{"errors":[{"status":"404","title":"The resource you are looking for '
 'doesn\'t exist."}]}').  This error may also be indicative of bad credentials when returned by DataCite.
...
```

## 🥼 Related Test Case(s)
doi-service #273 

## 🔁 : Related issues
N/A

---

## ➕ Additional Details
See regression tests for details at https://pds-jenkins.jpl.nasa.gov/view/int/job/pds-doi-console-regression-tests/job/develop/

### 📜 To Reproduce
1. Go to PDS-Jenkins Server and see failed [regression tests](https://pds-jenkins.jpl.nasa.gov/view/int/job/pds-doi-console-regression-tests/job/develop/).
2. Click "Build Now" and view console logs to see errors. 

### 🕵️ Expected behavior
Login succeeds and data import succeeds, no DataCite errors.

### 📚 Version of Software Used
doi-service 2.2.1+ (PyPi)

### 🩺 Test Data / Additional context
DEV password to datasite is used in the DOI Service INI file.

###  🏞Screenshots
NA, see link above.

### 🖥 System Info
 - OS: Ubuntu container
 - Browser: N/A
 - Version: 20.04

---

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
Overhauls configuration loading behaviour.

Previously, configurations were loaded from a handful of candidate locations which weren't robustly defined.  Now, there is a default config in the source code, and an optional location (matching that given in the doco) pulled from `sys.prefix` whose keys will override default values.  Environment variables will override both.

Default credential values for DataCite have been removed as these are a potential source of developer/user confusion.  If these values are read without being set in the user config or env vars, a descriptive exception will be thrown.

Adds comprehensive tests for `DOIConfigUtil` to prevent similar issues in future

## ⚙️ Test Data and/or Report
Unit tests pass locally.  Tested as wheel installation (where #371 arises) manually for a few actions, successfully.

## ♻️ Related Issues
Fixes #371 (probably.  99%.  unit tests aren't set up to be run against a wheel installation, but I've run a few manual tests)

### CODE REVIEW

**Comment by alexdunnjpl** [comment]
Need to sort out failing BIT.  I'm fuzzy on how the DataCite credentials are/were passed into the test environment, but I'll take a look at updating that tomorrow to supply them as env vars.

**Comment by alexdunnjpl** [comment]
Depends on https://github.com/NASA-PDS/doi-service/pull/373

File: docs/source/installation/index.rst [hunk 1]
```diff
@@ -75,9 +75,9 @@ following::
 Within the directory returned, create a file named ``pds_doi_service.ini``.
 
 In this file you can override any option set in the default configuration file
-``pds_doi_service/core/util/conf.ini.default`` found within the package. An
+``pds_doi_service/core/util/conf.default.ini`` found within the package. An
 example of this file may be found
-`here <https://raw.githubusercontent.com/NASA-PDS/pds-doi-service/main/src/pds_doi_service/core/util/conf.ini.default>`_.
+`here <https://raw.githubusercontent.com/NASA-PDS/pds-doi-service/main/src/pds_doi_service/core/util/conf.default.ini>`_.
```
Comment (nutjob4life) [inline docs/source/installation/index.rst:80]: Nit pick: [don't](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/) [hyperlink](https://heyoka.medium.com/dont-use-click-here-f32f445d1021) [the](https://ux.stackexchange.com/questions/12100/why-shouldnt-we-use-words-such-as-here-and-this-in-textlinks) [word](https://www.wordpress-web-designer-raleigh.com/2015/04/4-reasons-to-avoid-using-click-here-in-link-text/) [here](https://www.w3.org/QA/Tips/noClickHere) 😅

(Feel free to ignore this)

**Review by nutjob4life — APPROVED** [review]
Approved with the assurance that `OSTI_USER` and `OSTI_PASSWORD` env vars still work

**Comment by nutjob4life** [comment]
> > Approved with the assurance that `OSTI_USER` and `OSTI_PASSWORD` env vars still work
> 
> They won't (specifically, I removed their insertion into the config in #373 ), but my understanding is that OSTI is deprecated/obsolete. 

Then these files will need to be updated as well:

- `Jenkinsfile`
- `jenkins-config.yaml`
- `doi_service.env.in`

Since the PDS Exposition instance of the DOI service is using OSTI, right @jordanpadams?

File: docs/source/installation/index.rst (hunk 1 above)
Comment (alexdunnjpl) [reply docs/source/installation/index.rst:80]: Temporarily ignoring, but will go back and fix that

**Comment by jordanpadams** [comment]
@nutjob4life @alexdunnjpl no services should be using OSTI anymore. we no longer have access to their services.

**Comment by alexdunnjpl** [comment]
@nutjob4life on that basis, I'd cautiously suggest that we should be good to go - while you're correct that those files should be changed, I'd argue that should be part of complete OSTI excision (which we're waiting on @tloubrieu-jpl 's return to consider).

Basically the env vars' presence won't hurt anything, and I'd prefer not to touch anything unnecessary as I'm trying to sneak this in after the deadline for the current release.

**Comment by alexdunnjpl** [comment]
Rebased onto #373

**Pull request merged**